### PR TITLE
Fixes #35589 - Add repository subtab to CVV compare

### PR DESCRIPTION
**Pull request by sjha4**
#### To do

- [x] Tests

#### What are the changes introduced in this pull request?
Adds **Repository** comparison to CVV compare.
Screenshot from 2022-10-03 13-12-22

#### What are the testing steps for this pull request?
1. Add repos to CV and publish different versions.
2. Select 2 versions and click compare on versions tab
3. You should land on Repositories subtab of the compare screen.
4. Verify that comparison of repositories between versions works correctly along with Same/Different filters.

### CODE REVIEW

**Comment by sjha4** [comment]
[test katello]

**Comment by sjha4** [comment]
[test katello]

**Review by ianballou — COMMENTED** [review]
I noticed that comparing two versions without repositories leads to a blank page:
image

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -234,6 +234,51 @@ def index_relation_content_unit(query)
...
+      @versions = ContentViewVersion.readable.where(:id => params[:content_view_version_ids])
+      if @versions.count != params[:content_view_version_ids].uniq.length
+        missing = params[:content_view_version_ids] - @versions.pluck(:id)
+        fail HttpErrors::NotFound, _("Couldn't find content view versions '%s'") % missing.join(',')
+      end
+
+      archived_version_repos = Katello::Repository.where(:content_view_version_id => @versions&.pluck(:id))&.archived
+      repos = Katello::Repository.where(id: archived_version_repos&.pluck(:library_instance_id))
+      repos = repos.where(:root_id => @repo.root_id) if @repo
+      collection = restrict_comparison(repos, @versions, params[:restrict_comparison])
```
[the remark this answers is not in the document]
Comment (sjha4) [reply app/controllers/katello/api/v2/repositories_controller.rb]: updated this. Still using collection in the context of scoped_search and the response from the API cause that keeps it cleaner.

**Pull request merged**